fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindDynamicSplit.java

```diff
@@ -0,0 +1,83 @@
+import org.apache.doris.nereids.rules.RuleType;
+import org.apache.doris.nereids.trees.expressions.GreaterThanEqual;
+import org.apache.doris.nereids.trees.expressions.LessThanEqual;
+import org.apache.doris.nereids.trees.expressions.Slot;
+import org.apache.doris.nereids.trees.expressions.SlotReference;
+import org.apache.doris.nereids.trees.expressions.literal.Literal;
+import org.apache.doris.nereids.trees.plans.commands.info.SplitColumnInfo;
+import org.apache.doris.nereids.trees.plans.logical.LogicalCatalogRelation;
+import org.apache.doris.nereids.trees.plans.logical.LogicalDynamicSplit;
+import org.apache.doris.nereids.trees.plans.logical.LogicalFilter;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import org.apache.commons.lang3.Range;
+
+import java.util.List;
+
+/**
+ * This class defines the rule to bind a dynamic split operation in the query plan.
+ */
+public class BindDynamicSplit implements AnalysisRuleFactory {
+    @Override
+    public List<Rule> buildRules() {
+        return ImmutableList.of(
+
+                RuleType.BIND_DYNAMIC_SPLIT.build(logicalDynamicSplit().thenApply(ctx -> {
+                    if (!(ctx.root.child() instanceof LogicalCatalogRelation)) {
+                        return ctx.root;
+                    }
+
+                    LogicalDynamicSplit logicalDynamicSplit = ctx.root;
+                    Range range = logicalDynamicSplit.getRange();
+                    SplitColumnInfo splitColumnInfo = logicalDynamicSplit.getSplitColumnInfo();
+                    String splitColumnName = splitColumnInfo.getColumnName();
+                    TableName splitTable = splitColumnInfo.getTableNameInfo().transferToTableName();
+                    LogicalCatalogRelation catalogRelation = (LogicalCatalogRelation) ctx.root.child();
+                    TableIf table = catalogRelation.getTable();
+
+                    if (!table.getName().equals(splitTable.getTbl())) {
+                        return catalogRelation;
+                    }
+                    if (!table.getDatabase().getFullName().equals(splitTable.getDb())) {
+                        return catalogRelation;
+                    }
+                    if (!table.getDatabase().getCatalog().getName().equals(splitTable.getCtl())) {
+                        return catalogRelation;
+                    }
+                    SlotReference slotReference = null;
+                    for (Slot slot : catalogRelation.getOutputSet()) {
+                        if (slot.getName().equals(splitColumnName)) {
+                            slotReference = (SlotReference) slot;
+                        }
+                    }
+
+                    GreaterThanEqual greaterThanEqual = new GreaterThanEqual(slotReference,
+                            Literal.of(range.getMinimum()));
+                    LessThanEqual lessThanEqual = new LessThanEqual(slotReference, Literal.of(range.getMaximum()));
```

Comment:
will there be overlapping ranges in different batches?
for example [1, 100] separate to two parts, if you separate to [1, 50], [50, 100], the value '50' will be insert two times